app/services/kredits.js

```diff
@@ -283,29 +300,40 @@ export default Service.extend({
       });
   },
 
-  addProposal(proposal) {
-    const {
-      recipientAddress,
-      amount,
-      url
-    } = proposal.getProperties('recipientAddress', 'amount', 'url');
+  addProposal(attributes) {
+    debug('[kredits] add proposal', attributes);
+
+    let json = ContributionSerializer.serialize(attributes);
+    // TODO: validate against schema
 
     return this.get('ipfs')
-      .storeFile(proposal.serializeContribution())
-      .then(ipfsHash => {
+      .storeFile(json)
+      // Set ipfsHash
+      .then((ipfsHash) => {
```

Comment:
Just a formatting thing: if it's only one function argument, we don't need the parentheses around it. So it looks a bit cleaner and reads just a tiny little bit easier.